File: Remove All Adjacent Duplicates in String.cpp
#include <string>

class Solution {
public:
    string removeDuplicates(string s) {
        std::vector<char> newStr(s.length());
        int index = -1;
        for(char c: s){
            if(index >= 0 && c == newStr[index]){
                index--;
            } else{
                newStr[++index] = c;
            }
        }
        string finalStr(newStr.begin(), newStr.begin()+index+1);
        return finalStr;
        
    }
};

File: Remove All Adjacent Duplicates in String.java
import java.util.Stack;
class Solution {
    public String removeDuplicates(String s) {
        StringBuilder newStr = new StringBuilder();
        for(char c: s.toCharArray()){
            if(newStr.length() > 0 && newStr.charAt(newStr.length()-1) == c){
                newStr.deleteCharAt(newStr.length()-1);
            } else{
                newStr.append(c);
            }
        }
        return newStr.toString();
    }
}